### ISSUE TITLE
Routing: support restriction=no_entry

### ISSUE THREAD

**Issue by Neuwessi11** [osmandapp/OsmAnd#13049]
There are some situations where tagging with `restriction=no_entry` or one of the conditionals like `restriction:motorcar=no_entry` is the best possible solution.
So I want to ask for OsmAnd supporting this restriction.

Same with `restriction=no_exit`.

**Comment by Discostu36** [comment]
When implementing this, please keep in mind that this relation can have multiple `from` members.

**Comment by scaidermern** [comment]
Meanwhile there are >1334 [usages of `restriction=no_entry`](https://taginfo.openstreetmap.org/tags/restriction=no_entry). Maybe it's worth to add support for it?

**Comment by Peter0** [comment]
Imagine driving and then come to a situation where you must not enter a street due to the red white sign. 
Then one has no clue about the features further down in the street one is not allowed to enter. So in order to keep OSM up to date the only thing you could enter reliably is a no entry.

**Comment by Peter0** [comment]
My opinion: Assigning "nice to have" is an evidence that one may has lost the pretence considering Osmand's router seriously.

**Comment by scaidermern** [comment]
Indeed. Routing issues like this should get a higher priority. Navigation is the main task of this app and is even part of its name.

**Comment by Discostu36** [comment]
Keep in mind that a `no_entry` restriction can have more than one `from` member while a `no_exit` restriction can have more than one `to` members.

**Comment by Peter0** [comment]
> Keep in mind that a `no_entry` restriction can have more than one `from` member while a `no_exit` restriction can have more than one `to` members.

To me it sounds like that the conversion might be straight forward.

**Comment by Zirochkabila** [comment]
# Testing 1
`OsmAnd~ 5.0.0#4239m, Released: 2025-01-27`
`4.9.5.4954`

Route is built through [Restriction](https://www.openstreetmap.org/relation/5978540#map=19/48.249399/16.375256) 


<details><summary>Restriction</summary>
<table style="width:100%">
  <tr>
    <th></th>
    <th></th>
  </tr>
  <tr>
<th><img src="https://github.com/user-attachments/assets/a5413564-7615-4d6e-86da-317ac16944d8"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/user-attachments/assets/733a290f-d729-4fe6-a629-06019cd62f4e"  width="700" height="auto"/> </th>
  </tr>
</table>  
</details>

**Comment by vshcherb** [comment]
1. Maps should be regenerated - so wait till 5 Feb or regenerate
2. Multiple froms are not supported for restriction, only first from is supported.

**Comment by scaidermern** [comment]
> 2\. Multiple froms are not supported for restriction, only first from is supported.

But why? This requirement was mentioned several times.

Can't you simply duplicate the restriction for each from/to element to fulfill this requirement? Sounds rather easy to me.

**Comment by Zirochkabila** [comment]
# Testing 1
`OsmAnd~ 5.0.0#4259m, Released: 2025-01-29`
`4.9.5.4958`

**only first from is supported** - The expected route is built on the created map

[Wien.obf.zip](https://github.com/user-attachments/files/18590838/Wien.obf.zip)

Waiting for map updates


<details><summary>Wien</summary>
<table style="width:100%">
  <tr>
<th><img src="https://github.com/user-attachments/assets/9e7ffe9b-4a29-4100-9fce-0c36c16d46bd"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/user-attachments/assets/392b2e3d-1759-43ed-8be7-817c40c6fed1"  width="600" height="auto"/> </th>
  </tr>
</table>  
</details>
